src/app/components/shopping-list/shopping-list.component.ts

```diff
@@ -24,21 +24,32 @@ export class ShoppingListComponent implements OnInit {
 	constructor(public storage: StorageService) {}
 
 	async loadPantryList(): Promise<void> {
-		this.pantryList$ = await this.storage.get_all()
+		this.pantryList$ = await this.storage.get_all(true, false)
+		const adHoc = await this.storage.get_all(false, true)
+		adHoc.subscribe(adHocEntryList => {
+			this.inCart = adHocEntryList
+		})
 
 		this.pantryList$.subscribe(pantryList => {
-			this.runningLow = pantryList.filter(([, pantryItem]) => pantryItem.runningLow)
-			this.outOf = pantryList.filter(([, pantryItem]) => pantryItem.out)
+			this.runningLow = pantryList.filter(([, pantryItem]) => pantryItem.runningLow && !pantryItem.inCart)
+			this.outOf = pantryList.filter(([, pantryItem]) => pantryItem.out && !pantryItem.inCart)
+			this.inCart = this.inCart.concat(pantryList.filter(([, pantryItem]) => pantryItem.inCart))
```

Comment:
This probably indicates my lack of understanding of observables and `.subscribe()` but just to be clear, there is no chance of a race condition here, between 

```adHoc.subscribe(adHocEntryList => { this.inCart = adHocEntryList })```

and 

```this.pantryList$.subscribe(pantryList => { this.inCart = this.inCart.concat(pantryList.filter(([, pantryItem]) => pantryItem.inCart)) })```

such that the result of the subscription to `adHoc` occurs after the result of the subscription to `pantryList`, because we're `await`ing them both and so the subscriptions fire off immediately?

Reply:
ah gooooooood q - im not sure there is no chance.  will convert to a combo subscription